Tasks/InstallAppleCertificateV2/preinstallcert.ts

```diff
@@ -25,21 +25,35 @@ async function run() {
 
         let certPwd: string = tl.getInput('certPwd');
 
-        // get the P12 details - SHA1 hash and common name (CN)
-        let p12Hash: string = await sign.getP12SHA1Hash(certPath, certPwd);
+        // get the P12 details - SHA1 hash, common name (CN) and expiration.
+        const p12Properties = await sign.getP12Properties(certPath, certPwd);
+        let commonName = p12Properties.commonName;
+        const fingerprint = p12Properties.fingerprint,
+            notBefore = p12Properties.notBefore,
+            notAfter = p12Properties.notAfter;
+
         // give user an option to override the CN as a workaround if we can't parse the certificate's subject.
-        let p12CN: string = tl.getInput('certSigningIdentity', false);
-        if (!p12CN) {
-            p12CN = await sign.getP12CommonName(certPath, certPwd);
+        const commonNameOverride: string = tl.getInput('certSigningIdentity', false);
+        if (commonNameOverride) {
+            commonName = commonNameOverride;
         }
 
-        if (!p12Hash || !p12CN) {
-            throw tl.loc('INVALID_P12');
+        if (!fingerprint || !commonName) {
+            throw new Error(tl.loc('INVALID_P12'));
         }
-        tl.setTaskVariable('APPLE_CERTIFICATE_SHA1HASH', p12Hash);
+        tl.setTaskVariable('APPLE_CERTIFICATE_SHA1HASH', fingerprint);
 
         // set the signing identity output variable.
-        tl.setVariable('signingIdentity', p12CN);
+        tl.setVariable('signingIdentity', commonName);
+
+        // Warn if the certificate is not yet valid or expired. If the dates are undefined or invalid, the conditions will return false.
+        const now: Date = new Date();
+        if (notBefore > now) {
+            tl.warning(tl.loc('CertNotValidUntilWarning', notBefore));
+        }
+        if (notAfter < now) {
+            tl.warning(tl.loc('CertExpiredWarning', notAfter));
+        }
```

Comment:
Should we just fail the task if the cert is not valid? Seems reasonable since the build will fail downstream anyway.

Reply:
Please add an L0 test to validate task behavior on expired cert.

Reply:
Sounds reasonable as a default. A larger team might get some value out of seeing that tests still pass ahead of a signed build failing.

@DennisPan @meiravf Do you have any concerns with "Install Apple certificate" failing for expired certificates? Based on one test build, it looks like App Center is using v0 still.

Reply:
Yes we are still using v0. Would this change apply to v0 as well?

Reply:
I couldn't think of one. Fail fast means less build minutes used, so that should be good. Is there a place that we can look for these type of "breaking changes" when we decide to move to latest version?